PowerballApi/Repositories/PowerballRepository.cs

```diff
@@ -1,50 +1,90 @@
 ﻿namespace PowerballApi.Api.Repositories
 {
-	using System.Collections.Generic;
-	using Helpers.Cacher;
-	using Helpers.HttpHandler;
-	using Helpers.Parser;
-	using Models;
-
-	public class PowerballRepository : IPowerballRepository
-	{
-		public string CacheName { get; set; } = @"PowerballData";
-		public int DaysUntilStale { get; set; } = 1;
-		public string PowerballUrl { get; set; } = @"http://www.powerball.com/powerball/winnums-text.txt";
-
-		private readonly ICacher _cacher;
-		private readonly IHttpHandler _httpHandler;
-		private readonly IParser<PowerballSet, string> _parser;
-
-		public PowerballRepository()
-		{
-			_cacher = new Cacher();
-			_httpHandler = new HttpHandler();
-			_parser = new PowerballParser();
-		}
-
-		public PowerballRepository(ICacher cacher, IHttpHandler httpHandler, IParser<PowerballSet, string> parser)
-		{
-			_cacher = cacher;
-			_httpHandler = httpHandler;
-			_parser = parser;
-		}
-
-		public List<PowerballSet> Get()
-		{
-			var cache = _cacher.Get(CacheName);
-			if (cache != null)
-				return (List<PowerballSet>)cache;
-
-			return GetFreshDrawings();
-		}
-
-		private List<PowerballSet> GetFreshDrawings()
-		{
-			var file = _httpHandler.GetStringAsync(PowerballUrl).Result;
-			var results = _parser.Parse(file);
-			_cacher.Set(CacheName, results, DaysUntilStale);
-			return results;
-		}
-	}
+    using System.Collections.Generic;
+    using Helpers.Cacher;
+    using Helpers.HttpHandler;
+    using Helpers.Parser;
+    using Models;
+    using System;
+    using System.Linq;
+
+    public class PowerballRepository : IRepository<PowerballSet>
+    {
+        public string CacheName { get; set; } = @"PowerballData";
+        public int DaysUntilStale { get; set; } = 1;
+        public string PowerballUrl { get; set; } = @"http://www.powerball.com/powerball/winnums-text.txt";
+
+        private readonly ICacher _cacher;
+        private readonly IHttpHandler _httpHandler;
+        private readonly IParser<PowerballSet, string> _parser;
+
+        public PowerballRepository()
+        {
+            _cacher = new Cacher();
+            _httpHandler = new HttpHandler();
+            _parser = new PowerballParser();
+        }
+
+        public PowerballRepository(ICacher cacher, IHttpHandler httpHandler, IParser<PowerballSet, string> parser)
+        {
+            _cacher = cacher;
+            _httpHandler = httpHandler;
+            _parser = parser;
+        }
+
+        public List<PowerballSet> Get()
+        {
+            return GetDrawings();
+        }
+
+        public PowerballSet GetById(string id)
+        {
+            DateTime dateId;
+            if (!DateTime.TryParse(id, out dateId))
+                return null;
+
+            var data = GetDrawings();
+            try
+            {
+                return data.SingleOrDefault(p => p.Date == id);
+            }
+            catch (InvalidOperationException ex)
```

Comment:
If for some reason the data set is null, SingleOrDefault will throw an ArgumentException. We wouldn't catch that or any other exception because we're only looking for InvalidOperationException. Maybe add a default catch block?